src/Monolog/Registry.php

```diff
@@ -80,6 +98,17 @@ public static function hasLogger($logger)
     }
 
     /**
+     * Checks if such run-time option exists
+     *
+     * @param string $name key of the option
+     * @return bool
+     */
+    public static function hasOption($name)
+    {
+        return isset(self::$options[$name]);
```

Comment:
It is safer to do `array_key_exists($name, static::$options)` in case someone sets a value as `null` for some reason. Also, when accessing static member variables and properties, prefer `static::` to `self::`. The reason is that `static` supports late static binding whereas `self` does not.